src/app/clusters/scenes/SceneTableImpl.cpp

```diff
@@ -315,31 +449,91 @@ struct FabricSceneData : public PersistentData<kPersistentFabricBufferMax>
             VerifyOrReturnValue(this->Find(scene_id, scene.index) == CHIP_NO_ERROR, CHIP_NO_ERROR);
 
+            // Update the global scene count
+            GlobalSceneCount global_scene_count;
+            ReturnErrorOnFailure(global_scene_count.Load(storage));
+            global_scene_count.count_value--;
+            ReturnErrorOnFailure(global_scene_count.Save(storage));
+
             scene_count--;
             scene_map[scene.index].Clear();
-            ReturnErrorOnFailure(this->Save(storage));
+            err = this->Save(storage);
+
+            // On failure to update the scene map, undoes the global count modification
+            if (CHIP_NO_ERROR != err)
+            {
+                global_scene_count.count_value++;
+                ReturnErrorOnFailure(global_scene_count.Save(storage));
+                return err;
+            }
 
             err = scene.Delete(storage);
 
-            // On failure to delete scene, undoes the change to the Fabric Scene Data
-            if (err != CHIP_NO_ERROR)
+            // On failure to delete scene, undoes the change to the Fabric Scene Data and the global scene count
+            if (CHIP_NO_ERROR != err)
             {
+                global_scene_count.count_value++;
+                ReturnErrorOnFailure(global_scene_count.Save(storage));
+
                 scene_count++;
                 scene_map[scene.index] = scene.mStorageId;
                 ReturnErrorOnFailure(this->Save(storage));
+                return err;
             }
         }
         return err;
     }
+
+    CHIP_ERROR Load(PersistentStorageDelegate * storage) override
+    {
+        VerifyOrReturnError(nullptr != storage, CHIP_ERROR_INVALID_ARGUMENT);
+
+        uint8_t buffer[kPersistentFabricBufferMax] = { 0 };
+        StorageKeyName key                         = StorageKeyName::Uninitialized();
+
+        // Set data to defaults
+        Clear();
+
+        // Update storage key
+        ReturnErrorOnFailure(UpdateKey(key));
+
+        // Load the serialized data
+        uint16_t size  = static_cast<uint16_t>(sizeof(buffer));
+        CHIP_ERROR err = storage->SyncGetKeyValue(key.KeyName(), buffer, size);
+        VerifyOrReturnError(CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND != err, CHIP_ERROR_NOT_FOUND);
+        ReturnErrorOnFailure(err);
+
+        // Decode serialized data
+        TLV::TLVReader reader;
+        reader.Init(buffer, size);
+
+        err = Deserialize(reader);
+        // If Deserialize returns "buffer too small", the table in flash memory holds to many scenes (Can happend if
+        // max_scenes_per_fabric was reduced during an OTA). The fabric will then proceed to remove all scenes above the limit from
+        // the fabric
+        if (err == CHIP_ERROR_BUFFER_TOO_SMALL)
+        {
+            reader.Init(buffer, size);
+            ReturnErrorOnFailure(DeserializeAjust(reader, storage));
```

Comment:
So... why no have Deserialize do the "adjust" thing itself and have an extra out param that indicates it ended up deleting some scenes (so we need to call `this->Save(storage)`) instead of having two nearly-identical deserialize methods?

Reply:
Instead I added members to the Scene Fabric Data that get populated and checked by the Load() method.